web3/primitives.nim

```diff
@@ -39,6 +39,7 @@ type
     minLen: static[int] = 0,
     maxLen: static[int] = high(int)] = distinct seq[byte]
 
+  U8Quantity* = distinct uint8
```

Comment:
We don't need a new `U8Quantity`, only readValue/writeValue for uint8. n-j-s is smart enough today.
https://github.com/status-im/nim-web3/blob/6d46159259421f0105bbf86cfd4302e83c1fece6/web3/conversions.nim#L307 shows we can even get rid of Quantity if needed.

Reply:
> shows we can even get rid of Quantity if needed.

about that, we still lack the "flavor" option to disable the automatic serialization of integers and other primitives, so it's actually risky to rely on uint64

Reply:
okay, I think we need to implement https://github.com/status-im/nim-json-serialization/issues/121 first. looks not too difficult